client/src/pages/Register.js

```diff
@@ -0,0 +1,169 @@
+import React, { Component } from "react";
+import { withStyles } from "@material-ui/core/styles";
+//import { Router } from "react-router-dom";
+import NavBar from "../components/NavBar";
+import FormControl from "@material-ui/core/FormControl";
+import FormHelperText from "@material-ui/core/FormHelperText";
+import Input from "@material-ui/core/Input";
+import InputLabel from "@material-ui/core/InputLabel";
+import OutlinedInput from "@material-ui/core/OutlinedInput";
+import TextField from "@material-ui/core/TextField";
+import Button from "@material-ui/core/Button";
+import { flexbox } from "@material-ui/system";
+import Box from "@material-ui/core/Box";
+import Grid from "@material-ui/core/Grid";
+
+import axios from "axios";
+
+const registerPageStyle = theme => ({
+  landingContainer: {
+    //margin: theme.spacing.unit * 2
+  },
+  container: {
+    display: "flex",
+    flexWrap: "wrap"
+  },
+  textField: {
+    marginLeft: theme.spacing(1),
+    marginRight: theme.spacing(1)
+  }
+});
+
+class Register extends Component {
+  state = {
+    userName: " ",
+    userEmail: " "
+  };
+
+  //componentDidMount() {
+
+  // axios.get("api/users").then(res => console.log(res));
+  //}
+
+  /*
+  <form onSubmit={this.handleSubmit}>
+  Name:
+  <input
+    type="text"
+    value={userName}
+    name="userName"
+    onChange={this.handleChange}
+  />
+  <br />
+  <br />
+  Email:
+  <input
+    type="text"
+    value={userEmail}
+    name="userEmail"
+    onChange={this.handleChange}
+  />
+  <br />
+  <br />
+  <input type="submit" value="Sign up" />
+</form>
+*/
+
+  handleSubmit = async e => {
+    e.preventDefault();
+    let userData = {
+      name: this.state.userName,
+      email: this.state.userEmail
+    };
+    //make http post request to send name and email to server
+    await axios
+      .post("/api/users", userData)
+      .then(res => console.log("http result", res))
```

Comment:
Check the status of the response and only redirect the user if it's successful. Otherwise, throw an error or reject the promise

Reply:
Ok I will do that after I merge this branch, and resolve the difference made by Alain.

Reply:
Sounds good. In the feature you might wanna merge dev into yours first to solve conflicts :
`git checkout dev && git pull; git checkout branch-name; git merge dev` 
Then you can push your changes and request a review again.

Reply:
Oh shoot my bad I've already done it. I merged dev into my branch to fix the conflicts and already pushed the changes without review! I will fix this specific issue and will then request a review

Reply:
Sure, no worries